storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3UploadOutputStream.java

```diff
@@ -118,14 +125,21 @@ public void write(final byte[] b, final int off, final int len) throws IOExcepti
                 inputBuffer.position(positionAfterCopying);
 
                 if (!partBuffer.hasRemaining()) {
+                    if (uploadId == null){
+                        multiPartUpload = true;
+                        uploadId = createMultipartUploadRequest();
+                    }
                     partBuffer.position(0);
                     partBuffer.limit(partSize);
-                    flushBuffer(partBuffer.slice(), partSize);
+                    flushBuffer(partBuffer.slice(), partSize, true);
                 }
             }
         } catch (final RuntimeException e) {
-            log.error("Failed to write to stream on upload {}, aborting transaction", uploadId, e);
-            abortUpload();
+            if (multiPartUpload && uploadId != null) {
```

Comment:
I wonder if instead of a variable, `multiPartUpload` could be turned into a function `multiParrtUploadStarted() { return uploadId != null; }` so this validations could be simplified.

Reply:
Wouldn't the case 2 be covered by an exception? Maybe after creating a MPU add a check if the id is null and fail.

Reply:
@jeqo  already try to remove the field with new commit. you can help to check.  Thanks in advanced.